src/extension.ts

```diff
@@ -407,22 +407,25 @@ class DeviceTreeItem extends vscode.TreeItem {
         });
     }
 
-    public async printSysinfo() {
-        const messageBarItem = new StatusBarProgressionMessage("Grabbing ev3dev system info...");
-        try {
-            const sysinfo = await this.device.getSystemInfo();
-
-            output.appendLine("========== ev3dev-sysinfo ==========");
-            output.appendLine(sysinfo);
-            output.appendLine("");
-            output.show();
-
-            messageBarItem.finish("System info downloaded; see output window.");
-        }
-        catch(err) {
-            vscode.window.showErrorMessage("An error occurred while capturing system info: " + (err.message || err));
-            messageBarItem.finish();
-        }
+    public async showSysinfo() {
+        vscode.window.withProgress({
+            location: vscode.ProgressLocation.Window,
+            title: 'Grabbing ev3dev system info...'
+        }, async progress => {
+            try {
+                const sysinfo = await this.device.getSystemInfo();
+    
+                output.clear();
+                output.appendLine('========== ev3dev-sysinfo ==========');
+                output.appendLine(sysinfo);
+                output.show();
+    
+                toastStatusBarMessage('System info retrieved.');

```

Comment:
my inclination is to leave out the punctuation since we aren't using full sentences.